muikku-school-data-pyramus/src/main/java/fi/otavanopisto/muikku/plugins/schooldatapyramus/PyramusUserSchoolDataBridge.java

```diff
@@ -1367,15 +1364,19 @@ public BridgeResponse<StudentContactLogEntryRestModel> createStudentContactLogEn
     }
     BridgeResponse<StudentContactLogEntryRestModel> response =  pyramusClient.responsePost(String.format("/students/students/%d/contactLogEntries", studentId), Entity.entity(payload, MediaType.APPLICATION_JSON), StudentContactLogEntryRestModel.class);
 
-    boolean hasImage = false;
-    if (response.getEntity() != null) {
-      if (response.getEntity().getCreatorId() != null) {
-        response.getEntity().setCreatorId(toUserEntityId(response.getEntity().getCreatorId()));
-        
-        UserEntity userEntity = userEntityController.findUserEntityById(toUserEntityId(response.getEntity().getCreatorId()));
-        hasImage = userEntityFileController.hasProfilePicture(userEntity);
+    if (response.getEntity() != null && response.getEntity().getCreatorId() != null) {
+      response.getEntity().setCreatorId(toUserEntityId(response.getEntity().getCreatorId()));
+
+      boolean hasImage = false;
+      if (response.getEntity() != null) {
+        if (response.getEntity().getCreatorId() != null) {
+          response.getEntity().setCreatorId(toUserEntityId(response.getEntity().getCreatorId()));
+          
+          UserEntity userEntity = userEntityController.findUserEntityById(toUserEntityId(response.getEntity().getCreatorId()));
+          hasImage = userEntityFileController.hasProfilePicture(userEntity);
```

Comment:
Null check `userEntity`; will crash if, for some reason, `creatorId` doesn't result in `userEntity`